core/rawdb/schema.go

```diff
@@ -248,3 +249,48 @@ func accountTrieNodeKey(path []byte) []byte {
 func storageTrieNodeKey(accountHash common.Hash, path []byte) []byte {
 	return append(append(trieNodeStoragePrefix, accountHash.Bytes()...), path...)
 }
+
+// IsLegacyTrieNode reports whether a provided database entry is a legacy trie
+// node. The characteristics of legacy trie node are:
+// - the key length is 32 bytes
+// - the key is the hash of val
+func IsLegacyTrieNode(key []byte, val []byte) bool {
+	if len(key) != common.HashLength {
+		return false
+	}
+	return bytes.Equal(key, crypto.Keccak256(val))
+}
```

Comment:
We only use this during tests, right? Because it's one very expensive method with the hashing included. If only for tests, perhaps please add a comment to the method doc (since it's public) that it should not be used in production code.

Reply:
Interestingly, this will also return true for legacy code since that was also hash->code mapped. Wondering if it would be a good time to drop support for legacy codes and maybe have a startup check that chokes if you have such in your database (e.g. just look up the first deployed contract on mainnet and check it on startup if it's stored via hash or prefixed).

Reply:
It's also used by db inspector, so it's production code...

Reply:
db inspector is a debugging-tool, so not really _production_ production

Reply:
Out of curiosity, would the switch to PBSS be an opportune moment to prefix trie nodes in the database? I mean I guess sure, it would be a 1.4GB hit, but I'm unsure if it's not worth it long term to have trie nodes grouped together more meaningfully?

Reply:
But then is there a specific reason for the hashing here?

Reply:
Because in PBSS, the length of node path prefix is dynamic. It's possible that the path prefix is 31B, +1B database prefix, 32B totally which is exactly same as the hash based one. The additional hashing can help us to distinguish these two.

But, I am not strongly say that we should keep this hash operation. I am ok to remove it because it's only in database inspector and some "tiny fault" can be tolerant for sure.